### PR TITLE
Avoid unnecessary boxing in LogHelper

### PR DESCRIPTION
**Pull request by stephentoub**

### CODE REVIEW

**Review by jennyf19 — APPROVED** [review]
Logging is a huge issue for us, and we are looking to re-do it all-up.

File: src/Microsoft.IdentityModel.Logging/LogHelper.cs [hunk 1]
```diff
@@ -291,7 +289,7 @@ public static void LogVerbose(string message, params object[] args)
             if (IdentityModelEventSource.Logger.IsEnabled())
                 IdentityModelEventSource.Logger.WriteVerbose(message, args);
 
-            if (Enum.IsDefined(typeof(EventLogLevel), (int)EventLevel.Verbose) && Logger.IsEnabled((EventLogLevel)EventLevel.Verbose))
+            if (Logger.IsEnabled(EventLogLevel.Verbose))
                 Logger.Log(WriteEntry((EventLogLevel)EventLevel.Verbose, null, message, args));
```
Comment (keegan-caruso) [inline src/Microsoft.IdentityModel.Logging/LogHelper.cs]: Can we directly use EventLogLevel.Verbose here as well?

Comment (stephentoub) [reply src/Microsoft.IdentityModel.Logging/LogHelper.cs]: Yup, I missed that one, thanks (it won't change the code gen, but it'll certainly be cleaner)

Comment (stephentoub) [inline src/Microsoft.IdentityModel.Logging/LogHelper.cs]:
```suggestion
                Logger.Log(WriteEntry(EventLogLevel.Verbose, null, message, args));
```

**Review by keegan-caruso — APPROVED** [review]

**Pull request merged**